compiler/luci/pass/src/ForwardReshapeToUnaryOpPass.cpp

```diff
@@ -156,6 +183,55 @@ class ForwardReshape final : public luci::CircleNodeMutableVisitor<bool>
     return false;
   }
 
+  bool visit(luci::CircleMean *node)
+  {
+    luci::CircleReshape *reshape = nullptr;
+    luci::CircleConst *axis = nullptr;
+
+    reshape = dynamic_cast<luci::CircleReshape *>(node->input());
+    axis = dynamic_cast<luci::CircleConst *>(node->reduction_indices());
+
+    if (reshape == nullptr or axis == nullptr)
+      return false;
+
+    if (axis->dtype() != loco::DataType::S32)
+      return false;
+
+    // Should be scalar
+    if (axis->size<loco::DataType::S32>() != 1)
+      return false;
+
+    // axis value
+    int32_t axis_value;
+    axis_value = axis->at<loco::DataType::S32>(0);
+
+    if (axis_value < 0)
+      axis_value += static_cast<int32_t>(reshape->rank());
+
+    assert(axis_value >= 0);
+
+    if (node->keep_dims() != true)
+      return false;
+
+    auto *reshape_input = dynamic_cast<luci::CircleNode *>(reshape->tensor());
+    if (reshape_input == nullptr)
+      return false;
+
+    // reshape shouldn't change rank
+    if (reshape_input->rank() != reshape->rank())
+      return false;
+
+    assert(reshape_input->rank() > static_cast<uint32_t>(axis_value));
+
+    for (int32_t i = 0; i <= axis_value; ++i)
+    {
+      if (reshape_input->dim(i).value() != reshape->dim(i).value())
```

Comment:
maybe have to check `reshape_input->dim(i).known()` as input may have unknown shape.

Reply:
Done